src/consensus/aft/raft.h

```diff
@@ -1431,6 +1764,15 @@ namespace aft
         }
       }
 
+      if (async_exec.pending_cbs == 0)
+      {
+        return execute_append_entries_finish(confirm_evidence, r);
```

Comment:
I think it would be really good if we could separate the sync/async execution out of the consensus, where the implementation of the async flow does not belong, and have unit tests on the async flow to make sure it produces a valid order, which seem to be missing at the moment.

Reply:
i tried to doing this please have a look and tell me what you think